bento/src/main/java/com/yelp/android/bento/core/AsyncInflationBridge.kt

```diff
@@ -116,13 +114,150 @@ internal class AsyncInflationBridge @JvmOverloads constructor(
+            // view holders without blocking any more addComponent calls.
+            if (!belowTheFoldTriggered && !smartAsyncCacheEmptyOnStart &&
+                    SmartAsyncInflationCache.containsKey(name)) {
+                asyncInflateViewsIfPossible(aboveFold, false)
+            }
+        }
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private suspend fun asyncInflateViewsIfPossible(keys: List<Any>, isAboveTheFoldCall: Boolean) {
+        if (name.isEmpty()) return // There's no work to do.
+
+        // In theory asyncInflateViewsIfPossible should only be called two times.
+        // Once for above the fold, another for below the fold.
+        if (belowTheFoldTriggered) return
+
+        val mapToPrepare = SmartAsyncInflationCache[name] ?: mutableMapOf()
+        if (mapToPrepare.isEmpty()) return // There's no work to do.
+
+        if (!isAboveTheFoldCall) {
+            belowTheFoldTriggered = true
+        }
+
+        val keysToUse = if (isAboveTheFoldCall) keys else mapToPrepare.keys.subtract(keys)
+        keysToUse.forEach { viewHolderType ->
+            val numberOfViewsToInflate: Int = mapToPrepare[viewHolderType] ?: 0
+            val inflations: List<Deferred<Unit>> = (0 until numberOfViewsToInflate).map {
+
+                coroutineScope {
+                    async {
+                        val viewHolder = constructViewHolder(viewHolderType as Class<out ComponentViewHolder<Any?, Any?>>?)
+                        addViewHolder(viewHolder, viewHolderType as Class<out ComponentViewHolder<*, *>>)
+                        val (_, view) = BentoAsyncLayoutInflater.inflate(viewHolder, recyclerView, asyncInflaterDispatcher)
+                        viewMap[viewHolder] = view
+                    }
+                }
+            }
+            inflations.awaitAll()
+        }
+    }
+
+    /**
+     * The "BEST_GUESS" strategy for async view inflation. Inflates views for [component] if
+     * its count is non-zero and if doing so doesn't inflate more
+     * than [MAX_VIEWS].
+     */
+    private fun asyncInflateViewsBestGuess(component: Component, addComponentCallback: () -> Unit) {
+        inflationJobs[component] = launch(defaultBridgeDispatcher) {
+            var numberOfViewsToInflate = component.count
+            // If there's more than 10, chances are they'll be recycled and more will go unused.
+            // Not gonna lie. It's a guess and could be improved.
+            if (component.count > 10) {
+                numberOfViewsToInflate = component.count / 2
+            }
+            // If this amount would tip us over the MAX views value, only inflate enough
+            // to fill up to the max.
+            if (numberOfViewsToInflate + inflatedViewTracker.get() > MAX_VIEWS) {
+                numberOfViewsToInflate = (MAX_VIEWS - inflatedViewTracker.get())
+                        .coerceAtMost(numberOfViewsToInflate)
```

Comment:
This coerceAtMost seems weird since we're assigning it back to the value. Shouldn't we just do:
`numberOfViewsToInflate.coerceAtMost(MAX_VIEW - inflatedViewTracker.get())`?